### ISSUE TITLE
Canvas rendered at low-res on HiDPI displays

### ISSUE THREAD

**Issue by cHoco** [rhysd/neovim-component#10]
Hello, on HiDPI displays, like Apple retina displays the canvas is rendered at low resolution. When creating the canvas you should take into account the the devicePixelRatio to determine height and width and then properly scale when drawing onto it.

**Comment by rhysd** [comment]
You're right.  I forgot device pixel ratio...

Although I don't have retina display Mac, I think `<neovim-editor>` shows very small fonts (right?).  Default font size is 12px hence it shows 6px font because device pixel ratio is `2`.

I'll fix this.

**Comment by rhysd** [comment]
@cHoco 

If you can, screenshot is very helpful for me.

**Comment by cHoco** [comment]
It isn't rendered smaller, just with a lower resolution.
Here's a screenshot with neovim in terminal:
<img width="780" alt="screen shot 2015-12-30 at 15 39 00" src="https://cloud.githubusercontent.com/assets/85527/12052103/c9484c0a-af0b-11e5-8001-801c7e04c4b1.png">

and neovim in NyaoVim:
![screen shot 2015-12-30 at 15 38 54](https://cloud.githubusercontent.com/assets/85527/12052113/e765e788-af0b-11e5-9ceb-4fa94f6089c8.png)

As you can see NyaoVim looks a lot more *pixelated*

**Comment by rhysd** [comment]
Oh, I got it.  It seems that `<canvas>` requires twice size as its element size.

http://www.html5rocks.com/en/tutorials/canvas/hidpi/

Thanks!

**Comment by sethwoodworth** [comment]
Can confirm the above.  I have an HiDPI screen on linux I would be more than happy to test bugfixes on.

**Comment by rhysd** [comment]
I tried a bit but there are many fix points.  It may take more time to fix this...

**Comment by johnhamelink** [comment]
I reckon this is the only issue stopping me from trialling NyaoVim full time!

**Comment by rhysd** [comment]
I started to use MacBook Pro Early 2015 today and got the same blocker :trollface:

**Comment by rhysd** [comment]
I fixed this issue in HEAD as below.

- **before**

<img width="802" alt="2016-01-07 9 25 17" src="https://cloud.githubusercontent.com/assets/823277/12166408/1a5c10c8-b569-11e5-9ef0-d9b1036aa604.png">

- **after**

<img width="803" alt="2016-01-07 9 25 59" src="https://cloud.githubusercontent.com/assets/823277/12166412/2561371e-b569-11e5-8541-69e7636cb1ac.png">

At least in my environment (OS X 10.11.1, MacBook Pro Early 2015), it works well.  Could anyone also check HEAD?

```sh
$ git clone https://github.com/rhysd/neovim-component.git
$ cd ./neovim-component
$ npm run dep
$ npm run example
```

**Comment by johnhamelink** [comment]
Here's what I get when follow the above script:

<img width="1420" alt="screen shot 2016-01-07 at 15 09 59" src="https://cloud.githubusercontent.com/assets/101739/12173791/c81665b0-b550-11e5-9d33-810acf90b77b.png">

**Comment by metalelf0** [comment]
I get the same as @johnhamelink .

**Comment by rhysd** [comment]
@johnhamelink @metalelf0 

Ah, sorry, `build` directory was not created automatically :confounded: 
I fixed it in b5790f22ffc42c990bb468925621710c1fafa2d4.

**Comment by johnhamelink** [comment]
@rhysd I still needed to run `npm run build` before `npm run example` worked, but it does now work and it looks great :+1:

**Comment by rhysd** [comment]
Au, you're right, correct commands are below.

```sh
$ git clone https://github.com/rhysd/neovim-component.git
$ cd ./neovim-component
$ npm run dep
$ npm run build
$ npm run example
```

Thank you for confirming.

**Comment by cHoco** [comment]
Works great here :D

**Comment by rhysd** [comment]
@cHoco :+1:

**Comment by rhysd** [comment]
I'll close this because I released NyaoVim 0.0.10 including this fix.
Thank you, everyone!
